diff -urNp -x '*.orig' mhash-0.9.9.9.org/lib/Makefile.am mhash-0.9.9.9/lib/Makefile.am
--- mhash-0.9.9.9.org/lib/Makefile.am	2006-06-26 06:10:25.000000000 +0200
+++ mhash-0.9.9.9/lib/Makefile.am	2026-03-03 00:05:16.922939214 +0100
@@ -27,6 +27,8 @@ libmhash_la_SOURCES = $(FRAMEWORK) $(KEY
 
 libmhash_la_LDFLAGS = -version-info 2:1
 
+libmhash_la_CFLAGS = -fPIC
+
 EXTRA_DIST = libmhash.sym
 
 dist-hook:
